src/fileviews/ufileviewwithmainctrl.pas

```diff
@@ -1463,116 +1560,10 @@ procedure TFileViewWithMainCtrl.edtRenameExit(Sender: TObject);
 end;
 
 procedure TFileViewWithMainCtrl.edtRenameButtonClick(Sender: TObject);
-var
-  Key: Word = VK_RETURN;
 begin
-  handleKey(Sender, Key, []);
+  edtRenameOnKeyRETURN(Sender);
 end;
 
-{$IFDEF LCLCOCOA}
-procedure TFileViewWithMainCtrl.edtRenameKeyDown(Sender: TObject;
-  var Key: Word; Shift: TShiftState);
-begin
-  if key=VK_RETURN then
-    keyDownText:= originalText
-  else if key=VK_ESCAPE then
-    keyDownText:= edtRename.Text;
-end;
-
-procedure TFileViewWithMainCtrl.edtRenameOnChange(Sender: TObject);
-begin
-  originalText:= edtRename.Text;
-end;
-
-procedure TFileViewWithMainCtrl.edtRenameKeyUp(Sender: TObject;
-  var Key: Word; Shift: TShiftState);
-var
-  currentText: String;
-begin
-  currentText:= edtRename.Text;
-  if (Key=VK_RETURN) or (key=VK_ESCAPE) then
-  begin
-      if currentText<>keyDownText then
-      begin
-        // From the text has been changed,
-        // the EditButton is in the IME state, just return
-        Key:= 0;
-        exit;
-      end;
-  end;
-  handleKey( Sender, Key, Shift );
-end;
-{$ENDIF}
-
-procedure TFileViewWithMainCtrl.handleKey(Sender: TObject;
-  var Key: Word; Shift: TShiftState);
-var
-  NewFileName: String;
-  OldFileNameAbsolute: String;
-begin
-
-  case Key of
-    VK_ESCAPE:
-      begin
-        Key := 0;
-        edtRename.Visible:=False;
-        SetFocus;
-      end;
-
-    VK_RETURN,
-    VK_SELECT:
-      begin
-        Key := 0; // catch the enter
-
-        NewFileName         := edtRename.Text;
-        OldFileNameAbsolute := edtRename.Hint;
-
-        try
-          case RenameFile(FileSource, FRenameFile, NewFileName, True) of
-            sfprSuccess:
-              begin
-                edtRename.Visible:=False;
-                SetActiveFile(CurrentPath + NewFileName);
-                SetFocus;
-              end;
-            sfprError:
-              msgError(Format(rsMsgErrRename, [ExtractFileName(OldFileNameAbsolute), NewFileName]));
-          end;
-
-        except
-          on e: EInvalidFileProperty do
-            msgError(Format(rsMsgErrRename + ':' + LineEnding + '%s (%s)', [ExtractFileName(OldFileNameAbsolute), NewFileName, rsMsgInvalidFileName, e.Message]));
-        end;
-      end;
-
-{$IFDEF LCLGTK2}
-    // Workaround for GTK2 - up and down arrows moving through controls.
-    VK_UP,
-    VK_DOWN:
-      Key := 0;
-{$ENDIF}
-
-{$IFDEF LCLWIN32}
```

Comment:
This code should go to EditKeyDown.

Reply:
1. In the committed version, the code related to LCLGTK2  has been moved into EditKeyDown (line 297-300)
2. the code related to LCLWIN32 should no longer be needed in Lazarus2.2, i have tested it. of course I'm not 100% sure about it yet

```
{$IFDEF LCLGTK2}
     // Workaround for GTK2 - up and down arrows moving through controls.
     VK_UP,
     VK_DOWN:
       Key := 0;
 {$ENDIF}

{$IFDEF LCLWIN32}
```